modules/llrt_child_process/src/lib.rs

```diff
@@ -418,44 +625,204 @@ fn spawn<'js>(
 ) -> Result<Class<'js, ChildProcess<'js>>> {
     let args_0 = args_and_opts.first();
     let args_1 = args_and_opts.get(1);
-
     let mut opts = None;
 
     if args_1.is_some() {
         opts = args_1.and_then(|o| o.as_object()).map(|o| o.to_owned());
     }
 
-    let mut command_args = if let Some(args_0) = args_0 {
+    let mut command_args = match get_command_args(&ctx, args_0, &mut opts) {
+        Ok(Some(args)) => Some(args),
+        Ok(None) => None,
+        Err(err) => return Err(err),
+    };
+
+    let mut windows_verbatim_arguments: bool = get_windows_verbatim_arguments(opts.as_ref())?;
+
+    let cmd = get_cmd(
+        opts.as_ref(),
+        &mut command_args,
+        &mut windows_verbatim_arguments,
+        cmd,
+    )?;
+
+    let mut command = StdCommand::new(cmd.clone());
+    set_command_args(
+        &mut command,
+        command_args.as_ref(),
+        windows_verbatim_arguments,
+    );
+
+    let mut stdin = StdioEnum::Piped;
+    let mut stdout = StdioEnum::Piped;
+    let mut stderr = StdioEnum::Piped;
+
+    if let Some(opts) = opts {
+        get_gid(&opts, &mut command)?;
+        get_uid(&opts, &mut command)?;
+        get_cwd(&opts, &mut command)?;
+        get_env(&opts, &mut command)?;
+        get_stdio(&ctx, &opts, &mut stdin, &mut stdout, &mut stderr)?;
+    }
+
+    command.stdin(stdin.to_stdio());
+    command.stdout(stdout.to_stdio());
+    command.stderr(stderr.to_stdio());
+
+    #[cfg(unix)]
+    {
+        command.process_group(0);
+    }
+
+    //tokio command does not have all std command features stabilized
+    let mut command = Command::from(command);
+    ChildProcess::spawn(ctx.clone(), cmd, command_args, command.spawn())
+}
+
+fn exec_file<'js>(
```

Comment:
The refactor of `spawn` to accommodate `exec_file` is good. Let's split the helpers in another file (I would split functions together, the class on its own and the helpers together).